feat: add different llm provider options (google, mistral)

danfimov opened the pull request.

vas3k commented on docker-compose.build.yml:
```diff
@@ -9,7 +9,7 @@ services:
       - SELF_HOSTED_MODE=true
       - UPLOAD_PATH=/app/data/uploads
-      - DATABASE_URL=postgresql://postgres:postgres@postgres:5432/taxhacker
+      - DATABASE_URL=postgresql://postgres:postgres@postgres.taxhacker.orb.local:5432/taxhacker
```
This one should be reverted to original :)

vas3k reviewed (changes requested):
It's a good one! Couple more things:

1. Pls grep all the code to find where `settings.openai_api_key` also used — you'll find a bit more places to make changes. Otherwise it wont work. For example: https://github.com/search?q=repo%3Avas3k%2FTaxHacker%20settings.openai_api_key&type=code
2. In self-hosted mode we have a special "welcome page" where it asks user to provide OpenAI key. Now we have multiple providers, so it's better to add a dropdown there and change the form dynamically. This one: https://github.com/vas3k/TaxHacker/blob/main/app/(auth)/self-hosted/page.tsx#L47
3. Let's also add information about the Google provider to the README to emphasize that a self-hosted users might use one of the keys, either openai or google (there is an ENV vars example table)

danfimov commented:
<img width="1031" height="533" alt="Screenshot 2025-07-16 at 08 53 32" src="https://github.com/user-attachments/assets/ac80dc3e-7703-4ac3-b102-e2a716a8558c" />
<img width="836" height="660" alt="Screenshot 2025-07-16 at 08 53 47" src="https://github.com/user-attachments/assets/3ef959e6-4790-4ed7-8f48-2e3caecac471" />

Preview of changes in self-hosted setup page and settings page.

danfimov commented:
> 1\. Pls grep all the code to find where settings.openai_api_key also used — you'll find a bit more places to make changes. Otherwise it wont work.

I think now I found all the places that can cause an error. But code for three providers looks dirty. And there are a bunch of places with "if-else openai/google/mistral". I don't really know how to do it better, but for now it's pretty easy to shoot yourself in a foot during adding one more provider.

> 2\. In self-hosted mode we have a special "welcome page" where it asks user to provide OpenAI key. Now we have multiple providers, so it's better to add a dropdown there and change the form dynamically

I added select for different LLM providers on welcome page, but I'm not sure about design for this form. Maybe you have some thoughts about it?

> 3\. Let's also add information about the Google provider to the README to emphasize that a self-hosted users might use one of the keys

I added information about new env variables and ability to configure key/model in settings for users with self-hosted service in README. Is it enough?
